code/game/machinery/doors/brigdoors.dm

```diff
@@ -109,13 +98,75 @@
 
 /obj/machinery/door_timer/Destroy()
 	QDEL_NULL(Radio)
-	targets.Cut()
 	prisoner = null
+	targets.Cut()
 	return ..()
 
-//Main door timer loop, if it's timing and time is >0 reduce time by 1.
-// if it's less than 0, open door, reset timer
-// update the door_timer window and the icon
+/obj/machinery/door_timer/proc/update_record(mob/user)
+	var/datum/data/record/R = find_security_record("name", detaineename)
+	Radio.autosay("Detainee [detaineename] has been incarcerated for [timetoset/10] seconds for the charges of; '[detaineecrimes]'\
+	Arresting Officer: [user.name].[R ? "" : " Detainee record not found, manual record update required."]", name, "Security", list(z))
+
+	if(R)
+		prisoner = R
+		R.fields["criminal"] = "Incarcerated"
+		var/mob/living/carbon/human/M = user
+		var/rank = "UNKNOWN RANK"
+		if(istype(M) && M.wear_id)
+			var/obj/item/weapon/card/id/I = M.wear_id
+			rank = I.assignment
+		R.fields["comments"] = list()
+		R.fields["comments"] += "Autogenerated by [name] on [current_date_string] [worldtime2text()]<BR>Sentenced to [timetoset/10] seconds for the charges of \"[detaineecrimes]\" by [rank] [user.name]."
+		update_all_mob_security_hud()
+
+	return TRUE
+
+/obj/machinery/door_timer/proc/clearstringandtime()
+	detaineecrimes = ""
+	suggestedbrigtime = 0
+	timetoset = 0
+	for(var/datum/spacelaw/C in committedcrimes)
+		detaineecrimes += "#[C.name]	"
+		suggestedbrigtime += C.max_brig
+		timeset(suggestedbrigtime * 60)
+
+/obj/machinery/door_timer/proc/add_charge(mob/user)
+
+	if(detaineename == "")
+		add_name(user)
+
+	selecteddatum = new /datum/spacelaw

```

Comment:
@tigercat2000 They're not random?

The way I planned on having it work is by creating a new datum having the vars "name of crime" and "amount of time to add" and then add the datums to lists so they can be removed again. One datum for each crime?